### PR TITLE
IAM policies with s3 actions

### PR DESCRIPTION
**Pull request by ravindrachoudhari**
added test for IAM policy tests with s3 actions
tests executed on both ceph and cortx setup

### CODE REVIEW

File: s3tests_boto3/functional/test_iam.py [hunk 1]
```diff
@@ -413,3 +419,219 @@ def test_delete_user_policy_from_multiple_policies():
...
+    client.put_user_policy(PolicyDocument=policy_document_allow, PolicyName='AllowAccessPolicy',
+                           UserName=get_tenant_user_id())
+
+    s3_client = get_client()
+    s3_client.put_object(Bucket=bucket, Key='foo', Body='bar')
+    response = s3_client.get_object(Bucket=bucket, Key='foo')
+    body = response['Body'].read()
+    if type(body) is bytes:
+        body = body.decode()
+        print("decode")
```
Comment (apoorva-rao) [inline s3tests_boto3/functional/test_iam.py]: Maybe we can remove this print statement

**Review by apoorva-rao — APPROVED** [review]
@ravindrachoudhari LGTM, one print statement seems to have snuck in and it looks like it can be deleted. Added a comment inline for this.

File: s3tests_boto3/functional/test_iam.py (hunk 1 above)
Comment (ravindrachoudhari) [reply s3tests_boto3/functional/test_iam.py]: yes, Thank you.

File: s3tests_boto3/functional/test_iam.py [hunk 2]
```diff
@@ -413,3 +419,304 @@ def test_delete_user_policy_from_multiple_policies():
...
+    )
+    client.put_user_policy(PolicyDocument=policy_document_allow, PolicyName='AllowAccessPolicy',
+                           UserName=get_tenant_user_id())
+    s3_client = get_client()
+    response = s3_client.list_buckets()
+    for index in range(len(response['Buckets'])):
+        if bucket == response['Buckets'][index]['Name']:
+            bucket_listed = True
+            break
+        else:
```
Comment (seagate-sarang-sawant) [inline s3tests_boto3/functional/test_iam.py]: we don't need the else part.

File: s3tests_boto3/functional/test_iam.py [hunk 3]
```diff
@@ -413,3 +419,304 @@ def test_delete_user_policy_from_multiple_policies():
...
+        raise AssertionError("bucket is not listed")
+
+    s3_client.delete_bucket(Bucket=bucket)
+    response = s3_client.list_buckets()
+    bucket_listed = False
+    for index in range(len(response['Buckets'])):
+        if bucket == response['Buckets'][index]['Name']:
+            bucket_listed = True
+            break
+        else:
```
Comment (seagate-sarang-sawant) [inline s3tests_boto3/functional/test_iam.py]: We don't need the else part.

File: s3tests_boto3/functional/test_iam.py [hunk 4]
```diff
@@ -413,3 +419,304 @@ def test_delete_user_policy_from_multiple_policies():
...
+        raise AssertionError("deleted bucket is getting listed")
+    client.delete_user_policy(PolicyName='AllowAccessPolicy', UserName=get_tenant_user_id())
+
+
+@attr(resource='user-policy')
+@attr(method='s3 Actions')
+@attr(operation='Verify Deny Bucket Actions in user Policy')
+@attr(assertion='succeeds')
+@attr('user-policy')
+def test_deny_bucket_actions_in_user_policy():
```
Comment (seagate-sarang-sawant) [inline s3tests_boto3/functional/test_iam.py:469]: For a user can we add a scenario which allow bucket action, subsequently deny bucket action for same resource and allow bucket action at last and cleanup.

File: s3tests_boto3/functional/test_iam.py (hunk 2 above)
Comment (ravindrachoudhari) [reply s3tests_boto3/functional/test_iam.py]: yes, removed

File: s3tests_boto3/functional/test_iam.py (hunk 3 above)
Comment (ravindrachoudhari) [reply s3tests_boto3/functional/test_iam.py]: yes, removed

File: s3tests_boto3/functional/test_iam.py (hunk 4 above)
Comment (ravindrachoudhari) [reply s3tests_boto3/functional/test_iam.py:469]: This is getting covered in other tests
please check this PR https://github.com/Seagate/s3-tests/pull/4

**Review by ketanarlulkar — APPROVED** [review]

File: s3tests_boto3/functional/test_iam.py (hunk 4 above)
Comment (seagate-sarang-sawant) [reply s3tests_boto3/functional/test_iam.py:469]: ok

**Pull request merged**